src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,904 @@
+        /// </summary>
+        /// <returns>A string that contains the short date string representation of the current DateOnly object.</returns>
+        public string ToShortDateString() => ToString();
+
+        /// <summary>
+        /// Converts the value of the current DateOnly object to its equivalent string representation using the formatting conventions of the current culture.
+        /// The DateOnly object will be formatted in short form.
+        /// </summary>
+        /// <returns>A string that contains the short date string representation of the current DateOnly object.</returns>
+        public override string ToString() => ToString("d");
+
+        /// <summary>
+        /// Converts the value of the current DateOnly object to its equivalent string representation using the specified format and the formatting conventions of the current culture.
+        /// </summary>
+        /// <param name="format">A standard or custom date format string.</param>
+        /// <returns>A string representation of value of the current DateOnly object as specified by format.</returns>
+        public string ToString(string? format) => ToString(format, DateTimeFormatInfo.CurrentInfo);
+
+        /// <summary>
+        /// Converts the value of the current DateOnly object to its equivalent string representation using the specified culture-specific format information.
+        /// </summary>
+        /// <param name="provider">An object that supplies culture-specific formatting information.</param>
+        /// <returns>A string representation of value of the current DateOnly object as specified by provider.</returns>
+        public string ToString(IFormatProvider? provider) => ToString("d", provider);
+
+        /// <summary>
+        /// Converts the value of the current DateOnly object to its equivalent string representation using the specified culture-specific format information.
+        /// </summary>
+        /// <param name="format">A standard or custom date format string.</param>
+        /// <param name="provider">An object that supplies culture-specific formatting information.</param>
+        /// <returns>A string representation of value of the current DateOnly object as specified by format and provider.</returns>
+        public string ToString(string? format, IFormatProvider? provider)
+        {
+            if (format == null || format.Length == 0)
+            {
+                format = "d";
+            }
+
+            if (format.Length == 1)
+            {
+                switch (format[0])
+                {
+                    case 'o':
+                    case 'O':
+                        {
+                            Span<char> destination = stackalloc char[10];
+                            bool b = DateTimeFormat.TryFormatDateOnlyO(Year, Month, Day, destination);
+                            Debug.Assert(b);
+
+                            return destination.ToString();
```

Comment:
It'd be worth measuring to see which is faster, but consider:
```suggestion
return string.Create(10, this, (destination, value) =>
{
    bool b = DateTimeFormat.TryFormatDateOnlyO(value.Year, value.Month, value.Day, destination);
    Debug.Assert(b);
});
```
Same for the 'r' case below.

Reply:
will do.